src/test_workflow/benchmark_test/benchmark_test_base.py

```diff
@@ -0,0 +1,45 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+import subprocess
+from abc import ABC, abstractmethod
+from typing import Any
+
+from test_workflow.benchmark_test.benchmark_args import BenchmarkArgs
+
+
+class BenchmarkTestBase(ABC):
+    endpoint: str
+    security: bool
+    args: BenchmarkArgs
+    command: str
+    password: str
+
+    def __init__(
+            self,
+            endpoint: Any,
+            security: bool,
+            args: BenchmarkArgs,
+            password: str
+    ) -> None:
+        self.endpoint = endpoint
```

Comment:
only `args` can be initialized in this init method and remaining values can be initialized in respective command subclasses. You do not need to initialize any of these for `compare` command class, all the required parameters will be available from `args` in the super class.